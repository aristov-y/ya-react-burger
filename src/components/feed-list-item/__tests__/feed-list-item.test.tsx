import React from 'react';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import FeedListItem from '../index';
import { useSelector } from 'react-redux';

jest.mock('../../feed-list-ingredients', () => () => 'FeedListIngredients')

const onClick = jest.fn();

const store = {
  ingredients: {
    ingredients: [{
      _id: 'main01',
      price: 10
    }, {
      _id: 'main02',
      price: 13
    }, {
      _id: 'sauce01',
      price: 14
    }]
  }
}

beforeEach(() => {
  (useSelector as jest.Mock).mockImplementation(fn => fn(store));
});

describe('FeedListItem', () => {
  it('should render', () => {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<FeedListItem
        number={123}
        name={'Name'}
        createdAt={'2020-01-01T10:01:12'}
        ingredients={['main01', 'main01', 'sauce01']}
        onClick={onClick}
      />)
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
});
