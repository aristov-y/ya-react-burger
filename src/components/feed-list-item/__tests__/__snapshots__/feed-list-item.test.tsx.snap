// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedListItem should render 1`] = `
<div
  className="FeedListItem"
  onClick={[MockFunction]}
>
  <div
    className="FeedListItemId"
  >
    <span>
      #
      000123
    </span>
    <span
      className="text_color_inactive"
    >
      2020-01-01T05:01:12.000Z
    </span>
  </div>
  <div
    className="text_type_main-medium"
  >
    Name
  </div>
  <div
    className="FeedListItemIngredients"
  >
    FeedListIngredients
    <div
      className="FeedListItemPrice"
    >
      24
      CurrencyIcon
    </div>
  </div>
</div>
`;
